src/__tests__/labs/ViewLab.test.tsx

```diff
@@ -66,160 +66,149 @@ describe('View Lab', () => {
-    it('should have called the useUpdateTitle hook', async () => {
-      const expectedLab = { ...mockLab } as Lab
-      await setup(expectedLab, [Permissions.ViewLab])
-      expect(titleUtil.useUpdateTitle).toHaveBeenCalled()
-    })
-  })
-
   describe('page content', () => {
     it('should display the patient full name for the for', async () => {
       const expectedLab = { ...mockLab } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-      const forPatientDiv = wrapper.find('.for-patient')
-
-      expect(forPatientDiv.find('h4').text().trim()).toEqual('labs.lab.for')
+      setup(expectedLab, [Permissions.ViewLab])
 
-      expect(forPatientDiv.find('h5').text().trim()).toEqual(mockPatient.fullName)
+      await waitFor(() => {
+        expect(screen.getByText('labs.lab.for')).toBeInTheDocument()
+        expect(screen.getByText(mockPatient.fullName)).toBeInTheDocument()
+      })
     })
 
     it('should display the lab type for type', async () => {
       const expectedLab = { ...mockLab, type: 'expected type' } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-      const labTypeDiv = wrapper.find('.lab-type')
-      expect(labTypeDiv.find('h4').text().trim()).toEqual('labs.lab.type')
+      setup(expectedLab, [Permissions.ViewLab])
 
-      expect(labTypeDiv.find('h5').text().trim()).toEqual(expectedLab.type)
+      await waitFor(() => {
+        expect(screen.getByText('labs.lab.type')).toBeInTheDocument()
+        expect(screen.getByText(expectedLab.type)).toBeInTheDocument()
+      })
     })
 
     it('should display the requested on date', async () => {
       const expectedLab = { ...mockLab, requestedOn: '2020-03-30T04:43:20.102Z' } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-      const requestedOnDiv = wrapper.find('.requested-on')
-      expect(requestedOnDiv.find('h4').text().trim()).toEqual('labs.lab.requestedOn')
+      setup(expectedLab, [Permissions.ViewLab])
 
-      expect(requestedOnDiv.find('h5').text().trim()).toEqual(
-        format(new Date(expectedLab.requestedOn), 'yyyy-MM-dd hh:mm a'),
-      )
+      await waitFor(() => {
+        expect(screen.getByText('labs.lab.requestedOn')).toBeInTheDocument()
+        expect(
+          screen.getByText(format(new Date(expectedLab.requestedOn), 'yyyy-MM-dd hh:mm a')),
+        ).toBeInTheDocument()
+      })
     })
 
     it('should not display the completed date if the lab is not completed', async () => {
       const expectedLab = { ...mockLab } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-      const completedOnDiv = wrapper.find('.completed-on')
+      const { container } = setup(expectedLab, [Permissions.ViewLab])
 
-      expect(completedOnDiv).toHaveLength(0)
+      await waitFor(() => {
+        // TODO: make sure it's not loading
+        expect(container.querySelector('.completed-on')).not.toBeInTheDocument()
```

Comment:
Your thoughts on an alternative to querySelector?
.completed-on => implementation details?

Reply:
I don't know what is the type of element that will render so I don't know a better way to test it

Reply:
Yeah unfortunately to get some of the conversions to "just work" we are needing to take some RTL shortcuts.

Reply:
> Your thoughts on an alternative to querySelector?
> .completed-on => implementation details?

If you provide a debug() output or a playground URL we can take a look though.